code/modules/research/anomaly/anomaly_refinery.dm

```diff
@@ -0,0 +1,304 @@
+#define MAX_RADIUS_REQUIRED 20 //maxcap
+#define MIN_RADIUS_REQUIRED 4 //1, 2, 4
+/// How long the compression test can last before the machine just gives up and ejects the items.
+#define COMPRESSION_TEST_TIME (SSOBJ_DT SECONDS * 5)
+
+/**
+ * # Explosive compressor machines
+ *
+ * The explosive compressor machine used in anomaly core production.
+ *
+ * Uses the standard ordnance/tank explosion scaling to compress raw anomaly cores into completed ones. The required explosion radius increases as more cores of that type are created.
+ */
+/obj/machinery/research/anomaly_refinery
+	name = "anomaly refinery"
+	desc = "An advanced machine capable of implosion-compressing raw anomaly cores into finished artifacts. Also equipped with state of the art bomb prediction software."
+	circuit = /obj/item/circuitboard/machine/anomaly_refinery
+	icon = 'icons/obj/machines/research.dmi'
+	base_icon_state = "explosive_compressor"
+	icon_state = "explosive_compressor"
+	density = TRUE
+
+	/// The raw core inserted in the machine.
+	var/obj/item/raw_anomaly_core/inserted_core
+	/// The TTV inserted in the machine.
+	var/obj/item/transfer_valve/inserted_bomb
+	/// The timer that lets us timeout the test.
+	var/datum/timedevent/timeout_timer
+	/// Whether we are currently active a bomb and core.
+	var/active = FALSE
+	/// The message produced by the explosive compressor at the end of the compression test.
+	var/test_status = null
+	/// Determines which tank will be the merge_gases target (destroyed upon testing).
+	var/obj/item/tank/tank_to_target
+	
+	// These vars are used for the explosion testing and doesn't affect the core detonation.
+	/// Combined result of the first two tanks. Exists only in our machine, feel free to qdel.
+	var/datum/gas_mixture/combined_gasmix
+	/// Here for the UI, tracks the amounts of reaction that has occured. 1 means valve opened but not reacted.
+	var/reaction_increment = 0
+
+/obj/machinery/research/anomaly_refinery/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_INTERNAL_EXPLOSION, .proc/check_test)
+
+/obj/machinery/research/anomaly_refinery/Destroy()
+	UnregisterSignal(src, COMSIG_ATOM_INTERNAL_EXPLOSION)
+	return ..()
```

Comment:
```suggestion
```
Signals are unregistered by default by/from objects being destroyed.
Good thinknig, but you only have to unregister from things that are not being deleted.